docs/user-guide/deployments.md

```diff
@@ -454,6 +454,125 @@ nginx-deployment-3066724191   0         0         1h
 Note: You cannot rollback a paused Deployment until you resume it.
 
 
+## Failed Deployment
+
+There will be times your Deployment will get stuck trying to deploy its newest ReplicaSet without ever completing,
+due to insufficient quota, readiness probe failures, image pull error, insufficient permissions, limit ranges, or any
+application runtime misconfiguration. In those cases you can specify a timeout parameter in the spec
+(`spec.progressDeadlineSeconds`) that denotes the seconds you want to wait for your Deployment to report any progress.
+Once the deadline is exceeded, meaning your Deployment will not report any progress for more than progressDeadlineSeconds,
+the deployment controller will add a Progressing condition with Status=False and Reason=ProgressDeadlineExceeded in the
+status of the Deployment.
+
+Note that at the moment, Kubernetes will do nothing more than just report with a `ProgressDeadlineExceeded` condition.
+In the future, once we support automatic rollbacks and users have opted-in, the deployment controller will rollback as
+soon as a failure condition is noticed.
+
+Another thing to note is transient errors eg. your rollout times out and a `ProgressDeadlineExceeded` condition is set to
+your Deployment because don't have enough quota. All you need is to scale down your Deployment, or scale down other Pods
+you may be running, or increase quota in your namespace, either by asking your administrator or by doing it yourself assuming
+you have the right permissions. Once the deployment controller notices that your Deployment is complete, it will transition
+the status of the condition into a succeeded one (Status=True, Reason=NewReplicaSetAvailable).
+
+A complete Deployment for the controller is one that: 
+* has minimum availability, meaning its available replicas are more than those required by the Deployment strategy (`status.
+availableReplicas >= spec.replicas - maxUnavailable` where `maxUnavailable` is the maximum unavailable replicas that a
+Deployment allows at any point in time. It depends on the deployment strategy: Recreate deployments don't use it whereas
+Rolling deployments specify it in `spec.strategy.rollingUpdate.maxUnavailable`).
+* has all of its replicas updated to the latest revision (`spec.replicas == status.updatedReplicas`).
```

Comment:
The first point definitely applies for recreate deployments - in order to have a complete deployment you have to have all pods available, right?